pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -818,14 +817,14 @@ func (cb *ClusterBuilder) buildAutoMtlsSettings(
 		}
 		// Update TLS settings for ISTIO_MUTUAL. Use client provided SNI if set. Otherwise,
 		// overwrite with the auto generated SNI. User specified SNIs in the istio mtls settings
-		// are useful when routing via gateways. Use Service Acccounts if Subject Alt names
+		// are useful when routing via gateways. Use Service Accounts if Subject Alt names
 		// are not specified in TLS settings.
 		sniToUse := tls.Sni
 		if len(sniToUse) == 0 {
 			sniToUse = sni
 		}
 		subjectAltNamesToUse := tls.SubjectAltNames
-		if len(subjectAltNamesToUse) == 0 {
+		if subjectAltNamesToUse == nil {
```

Comment:
This could change behavior. Is it intended?

In general I am nervous about `nil` check for list since it means `[]` and `nil` are different..

Reply:
@howardjohn I meant to change this, I feel if user provided [], it means he wants no SAN.